nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -138,7 +138,16 @@ fi
 # Ask the user to set a root password, but only if the passwd command
 # exists (i.e. when mutable user accounts are enabled).
 if [[ -z $noRootPasswd ]] && [ -t 0 ]; then
+    set +e
     nixos-enter --root "$mountPoint" -c '[[ -e /nix/var/nix/profiles/system/sw/bin/passwd ]] && echo "setting root password..." && /nix/var/nix/profiles/system/sw/bin/passwd'
+    exit_code=$?
+    set -e
+
+    if [[ $exit_code != 0 ]]; then
+        echo "Setting a root password failed with the above printed error."
+        echo 'You can set the root password manually by executing `nixos-enter` and then runnig `passwd` in the shell of the new system.'
```

Comment:
```suggestion
        echo 'You can set the root password manually by executing `nixos-enter` and then running `passwd` in the shell of the new system.'
```

Reply:
Is this always true, or do they need to pass `--root $mountPoint`, and what happens if passwd doesn't exist?

Reply:
You're right, the mount point should be shown as well. And indeed, `nixos-enter` also breaks if the `[[ -e ... ]]` check breaks. I'll fix that :)